Update tests.yml to new layout

Luyunmt opened the pull request:
We are running live Tests against other clouds like US Gov and Azure China Cloud. The goal is to check whether new azure sdk package work with other clouds or not.

Using this RR to trace and fix issues we encounter in the tests. Will close it once we finish all the tests and send PR per services.

Updated list:
Text analytics
Core
Extensions
Event hubs
Identity
key vault
Search
Storage account
Form recognizer
Service bus
Event grid
Tables

benbp committed:
Diff sovereign cloud live test updates, merged with cloud configuration

benbp committed:
Diff sovereign cloud live test updates, merged with cloud configuration

benbp committed:
Diff sovereign cloud live test updates, merged with cloud configuration

benbp committed:
Use test keyvault subscription configuration values

benbp committed:
Use test keyvault subscription configuration values

benbp committed:
Use test keyvault subscription configuration values

benbp committed:
Default to true when live test platform does not have SupportedClouds override

benbp committed:
set env vars with no test resources

benbp committed:
set env vars with no test resources

pavelprystinka committed:
[Communication] - PhoneNumberAdministrationClient - adding live tests (#16962)

pakrym committed:
Update storage generator to suppress FxCop issues (#17181)

JoshuaLai committed:
[HMACAuthenticationPolicyTest] added new not null check for authValue (#17178)

heaths committed:
Enable code coverage (#17067)

* Use VSTest integration for Coverlet

Fixes #15231. Relates to #14427.

* Do not ignore CompilerGeneratedAttribute

See https://github.com/coverlet-coverage/coverlet/issues/794#issuecomment-612067552

* Enable code coverage for PRs

* Publish code coverage even for failed tests

* Enable code coverage reports

Resolves #14427 by limiting reports to a single service directory (CIs) or per test project (dev environments). For CIs, full coverage reports of everything build will be uploaded. For dev environments, a summary HTML file will be output to test projects' TestResults directories.

* Fix filefilters path for CI

* Use absolute path for filefilters

Same as dev support; reportgenerator documentation is not clear on when absolute or relative paths are required.

* Use absolute path for reports

* Update report directory to limit globbing

* Disable code coverage of track 2 mgmt

Opened #17090 to track re-enabling once improved.

* Generate report before uploading test results

The exact same commands are working locally using the same versions of ReportGenerator, sans running the test publishing executable. There's also extra coverage files showing up that I'm wondering if it's responsible.

* Temporarily upload all code coverage artifacts

* Split between props and targets again

At one point, this mostly worked. Going back to how I was split between props and targets before to see if that makes a difference.

* Collect more information

* Always define CoverletGetPathMap

@clairernovotny recommended important some changes. These are working locally when simulating a CI, so 🤞.

* Resolve PR feedback

Fix extra (temporary) logging as well.

* Use different variable to detect CI

ContinuousIntegrationBuild wasn't defined for test projects.

* Replace curly braces with Of in file names

Fixes #17164

* Removing extra logging

* Renames files with curly braces

Fixes #17164

* Resolve PR feedback

* Do not reformat HTML coverage report

pavelprystinka committed:
fix live test issue (#17200)

pakrym committed:
Install dotnet when running smoke tests (#17208)

Commit:
ADT: add info about object serializer override (#17199)

JoshLove-msft committed:
Add blurb about dist tracing (#17059)

- Addresses https://github.com/Azure/azure-sdk-for-net/issues/16238
- Update LineCounter sample app to show usage of Event Grid
- Update LineCounter README to include a screenshot of Azure Monitor
- Update Distributed Tracing section in Diagnostics readme to link to LineCounter.
- Also add in DI extension methods.

Tianyun-G committed:
Fix EventGrid Readme issue (#17209)

* Fix EventGrid Readme issue

* Update README.md

pavelprystinka committed:
do not check if there any aquired numbers. Will cover this check in the sample tests (#17212)

kasobol-msft committed:
log azurite stderr. (#17230)

chamons committed:
[CODEOWNERS] Add chamons to synapse (#17238)

heaths committed:
Fix doc comment for CertificatePolicy.Default (#17231)

benbp committed:
Temporarily disable live test weekly allow-all filter (#17236)

maririos committed:
ConvertPrebuiltOutputToRecognizedForms moved to ClientCommon (#17185)

christothes committed:
Properly create secondary endpoint Uri for Azurite endpoints (#17246)

fixes #17215

maririos committed:
[FR] Add AAD tests for every endpoint (#17191)

* add tests for every endpoint

* remove playback

* custom forms with and withouth labels for aad

pakrym committed:
Use BlobsCheckpointStore in scale monitor (#17241)

greenms committed:
Update Redis Enterprise to match Updated Swagger (#17214)

* Add support for Redis Enterprise.

* Add SessionRecords from successful tests.

* Add redisenterprise pipeline.

* Updated SDK to match updated Swagger.

jsquire committed:
[Event Hubs Client] Live Test Fixes (#17261)

The focus of these changes is to fix a couple of race conditions causing
some tests to be flaky and to adjust the scope creation approach used for
snippet test.  The global setup/cleanup was causing occasional long delays
that appeared to be hangs during nightly runs.

pavelprystinka committed:
[Communication] - PhoneNumberClient - Adding samples for phone number administration (#16935)

* inroduce phone number addministration tests

* Samples are created

* Improve sanitization in tests

* Improve sanitization in tests

* Update test records

* Ignore phone numbers live test for not playback if INCLUDE_PHONENUMBER_LIVE_TESTS env var is not True

* Update readme

* Update readme

* samples wording update

* rename test

* addressing tests code formatting

* addressing tests code formatting

* add assertion to sample test

* add release phone number method for single number

* add release phone number method for single number - update api

* address PR comments (sample wording and renames)

* update sample wording and some variable names

isaacbanner committed:
Send metadata header on IMDS probe request. (#15875)

* Call imds instance endpoint instead of token endpoint to probe for imds, extend timeout

* Update msi token tests to distinguish between which imds endpoint is being called

* Fix comment and visibility in test

* Change per stpetrov

* re-trigger checks

* Add metadata header to probe request

JoshLove-msft committed:
CloudNative.CloudEvents bridge library (#17234)

* CloudNative.CloudEvents bridge library

* PR fb

* Fix NUnit casing

* Export API

Luyunmt committed:
Update test.yml

Luyunmt committed:
Update test.yml

Luyunmt committed:
update test.yml

Luyunmt committed:
update test.yml

Luyunmt committed:
Update test.yml

Luyunmt committed:
Update test.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

Luyunmt committed:
Update tests.yml

heaths committed:
Add Mohit to Search owners list (#17275)

pakrym committed:
Remove abstraction layers from EventHubs WebJobs extensions (#17271)

There are some layers in EventHubs WebJobs extensions that were the artifact of Track 2 refactoring. Remove them to simplify the code a bit.

Major changes:
* Merge `Processor` and `EventProcessorHost`
* Merge `ProcessorPartitionContext` and `Partition`, rename to `EventProcessorHostPartition`
* Remove ICheckpointer

maririos commented:
Hi @Luyunmt . couple of questions:
- Will all this different clouds be tested in the weekly pipelines? or in all of our schedule pipelines
- Curious about the changes for Text analytics as they are mentioned in the description but I don't see the file changes.

amnguye committed:
Fix to be able to CanGenerateSas and GenerateSas from storage clients generated by GetClient (#17232)

scbedd committed:
Support moniker specific output folders (#17295)

* updating CI config with moniker specific readme/service output

JoshLove-msft committed:
Update to latest AMQP package version (#17290)

heaths committed:
Do not generate reports with no coverage (#17293)

Fixes #17286

sima-zhu committed:
Delete redundant files and point to common scripts (#17292)

seanmcc-msft committed:
Added support for AccountInfo.IsHierarchicalNamespaceEnabled (#17285)

Commit:
Fix template path for tests.yml (#17198)

Commit:
Add a default value for environmentName (#17309)

Xynoclafe committed:
Resources SDK version 2020-10  (#17163)

* Generated SDK and re-recorded tests

* rerecord remaining tests

* Updated version and release notes

* regenerate from Swagger. Fixed possible breaking change

danicole committed:
ClientCertificateAzureServiceTokenProvider dispose of certificate obj… (#17266)

…ects

Properly dispose the X509Certificate2 objects after use to prevent temporary private key files to filling up the disk.
Unless properly dispoed, they fill up the space between garbage collections and stay permanently in case of process kills.

pakrym committed:
Add token credential auth support (#17308)

chamons committed:
[synapse] Trim resource name to 15 char limit in provisioning (#17310)

- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17237

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1233 (#17314)

* reformatted parameter arguments to both update-docs scripts. added parameter requirements and renamed CIRepository to DocRepoLocation which is a lot clearer. Added additional CIConfig parameter to update-docs-metadata to support specific write locations per moniker.

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>

pavelprystinka committed:
[Communication] - PhoneNumberAdministrationClient - fix next page (#17283)

Fix communication admin Pageable results issue not pulling next page

azure-sdk committed:
Update AutoRest C# version (#17319)

ellismg committed:
[EventGrid] Small README.md updates (#17313)

Endpoint is a URI, let's make the construction explicit in our samples
instead of just a plain string so customers starting from these samples
don't hit a type error right away.

kasobol-msft committed:
[Storage] Increase network timeout in tests. (#17323)

Sometimes tests experience slow transfer rate which makes tests pushing larger payloads flaky.
This is attempt to give them more time to complete transfer and see if this resolves that problem.

ellismg committed:
[EventGrid] Add a cross reference to the docs (#17336)

From our user studies we've seen cases where folks correctly find
`EventGridSharedAccessSignitureCredential` but don't figure out how to
construct one.

We already have pointers on some of the constructors for
`EventGridPublisherClient` that take `EventGridSharedAccessSignature`'s
but we can also add a reference on the type itself.

christothes committed:
Various renaming for x-language consistency (#17320)

* Various renaming for x-language consistency

pakrym committed:
Enable multiple blank lines StyleCop rules (#17318)

heaths committed:
Enable track 2 mgmt projects by default (#17324)

Resolves #17090 - at least for re-enabling mgmt-plane. Still needs coverage improvement but that's being handled separately.

ShivangiReja committed:
Sdk automation script (#17193)

SDK automation script

christothes committed:
Fix sync batch method coverage (#17343)

* fix sync batch method coverage

christothes committed:
re-enable InheritDoc (#17259)

* re-enable inheritdoc and fix non-generated comment warnings

JoshLove-msft committed:
Clean up docs (#17342)

maririos committed:
add AAD and improve polling for playback (#17296)

maririos committed:
[TA] No locale in links and add functionalities to package descriptions (#17312)

* remove local from links and modify package description

pavelprystinka committed:
Make pageable results internal (#17348)

maririos committed:
[TA] Update service version and some cleanup (#17347)

* update service version

* run export

kasobol-msft committed:
[Storage] Bump network timeout, add progress logging. (#17350)

- Bump network timeout to 400s to see if it's enough for worst case agent-network-storage account combo
- add progress logging to flaky tests to see if it hangs.

amnguye committed:
STG 75 (#17242)

maririos committed:
[FR] Update docstrings (#17345)

* first rounds:

* improve docstring and cleanup some code

* PR feedback

jsquire committed:
[Event Hubs Client] Credential Refresh Buffer Tweak (#17330)

The focus of these changes is to add more buffer when computing the
time when a credential should be renwed.  We've recently begun to see
the occasional credential expiration infrequently during nightly runs.

These tweaks will allow for a greater period clock skew by eagerly refreshing
the token 10 minutes before the client belives it expires.  The minimum interval
for refreshing has also been reduced in order to guard against short lived
credentials.

jsquire committed:
[Event Hubs Client] Enable Idempotent Producer Tests (#17316)

The focus of these changes is to enable the Live tests for the Idempotent
Producer feature now that the service team has confirmed that it has rolled
out to all Azure clouds in our test matrix.

jsquire committed:
[Data Lake Performance] Test Scenarios for T2 (#17307)

The focus of these changes is to create the initial set of performance
test scenarios for the v12.x (track two) Data Lake client library.

abatishchev committed:
Replaced hardcoded true/false with TrueString/FalseString (#17359)

jsquire committed:
[Event Hubs Client] Sample ReadMe Updates (#17315)

The focus of these changes is to fix the instructions for adding packages
since there are references to API changes from the beta in the sample code.

maririos committed:
enable AAD in test resources (#17351)

amnguye committed:
Make storage mock clients able to generate sas (#17349)

seanmcc-msft committed:
Prepare storage for release (#17382)

LarryOsterman committed:
Create Azure.Security.Attestation Track 2 Preview SDK  (#17159)

* Created Initialcut at GA API version

* Checkpoint

* Checkpoint

* Implemented most of the core attestation service APIs.

* Implemented V1 of all of MAA API Surface

* Converted TPM request and response to ReadOnlyMemory

* Another round of API review feedback

* Updated readme.md and changelog.md a small bit; corrected a couple of API review feedback items

* Somehow nunit got added to the test dependencies; removed it

* Added snippets to readme.md

* Fixed typo in tests.yml; Fleshed out readme.md a bit

* Use ClientSecretCredentials not DefaultProviderCredentials for tests

* Two test classes were still flagged as being live

* Ran Export-API.ps1

* Updated root cl for autorest

* Updated session records

* Added new SDK to ci.yml

* Fixed several static analysis errors

* Use TestEnvironment.Credential for credentials

* Try a different way of generating the key

* Updated session records

* Updated session records with policy document that does not contain CRLF characters

* Pull request feedback changes

* Updated changelog

* More Pull Request feedback

* Added assert that the number of signing certificates is at least 1

* Use Argument.AssertNotNullOrEmpty instead of Argument.IsInRange for array check

kasobol-msft committed:
[Storage] Live tests - Wait for RBAC replication. (#17385)

It may take up to 5 minutes for RBAC (AAD) role assignments to replicate. This makes live tests flaky.
The short term solution is to add "sleep" after test resources provisioning.

Long term solution to be found here https://github.com/Azure/azure-sdk-for-net/issues/17384

chamons committed:
[Synapse] Update swagger

- Bump to eea7c0141c0b7ad9f66f8ba06560b549c6b3b014

pakrym committed:
Add legacy checkpoint reading support to EventHubs (#17335)

JoshLove-msft committed:
Update SB migration guide (#17357)

christothes committed:
diable inheritdoc again while troubleshooting failures (#17390)

* diable inheritdoc again while troubleshooting failures

sima-zhu committed:
Mash up the index.html and its relative links (#17267)

azure-sdk committed:
Increment version for storage releases (#17389)

* Increment package version after release of Azure.Storage.Common

* Increment package version after release of Azure.Storage.Blobs.Batch

* Increment package version after release of Azure.Storage.Blobs

* Increment package version after release of Azure.Storage.Files.Shares

* Increment package version after release of Azure.Storage.Queues

* Increment package version after release of Azure.Storage.Files.DataLake

* Increment package version after release of Azure.Storage.Blobs.ChangeFeed

maririos committed:
[FR] Update service version, TODO cleanup, and add missing xxOperation tests (#17387)

* service version and easy TODO cleanup

* parameter validation avoiding breaking changes

* add missing tests

* update docstring

* remove checks that throw in other methods

pakrym committed:
Enable legacy checkpoint reading in WebJobs (#17392)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17034

Also include name casing in tests.

JoshLove-msft committed:
Session receiver fix (#17398)

* Don't mutate SessionReceiverOptions

* Add topic/subscription coverage

* Pass sessionId

* Make internal param required

sima-zhu committed:
Fixed the navigation toc (#17400)

benbp committed:
Skip canary for search weekly live tests (#17391)

suhas92 committed:
[Text Analytics] update samples for healthcare (#17407)

* update samples for healthcare

* run updatesnippits

Co-authored-by: Suhas Mehta <suhas.mehta@microsoft.com>

pakrym committed:
GeoJSON API adjustments (#15836)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1257 (#17408)

* Make up the css using for .net sdk github io pages

* Remove index.html

* Make changes to js

Co-authored-by: Sima Zhu <sizhu@microsoft.com>

sima-zhu committed:
Simplify the root toc page. (#17413)

LarryOsterman committed:
Updated changelog version (#17406)

Updated changelog version

kasobol-msft committed:
[Storage][Webjobs] log if azurite hangs or dies. (#17419)

In this PR.
- if azurite died within 15s log this
- otherwise kill it and force output streams flush.

mitchdenny committed:
Enable 1ES hosted pools. (#17360)

* Enable 1ES hosted pools.

Luyunmt committed:
Merge branch 'benbp/sov-merge-testing' into benbp/sov-merge-testing

Luyunmt committed:
change extension

Luyunmt committed:
change extension

Luyunmt committed:
Merge branch 'benbp/sov-merge-testing' of https://github.com/Luyunmt/azure-sdk-for-net into benbp/sov-merge-testing

jsquire committed:
[Event Hubs Client] Model Factory (#17405)

The focus of these changs is to implement the Azure SDK model factory
pattern, to offer better support for mocking and testing efforts.

pakrym committed:
Map old configuration settings to new options type (#17416)

christothes committed:
Handle multi-part paths for storage folder Uris for backup/restore (#17415)

* restore operation folder parsing improvements

christothes committed:
bump InheritDoc to 1.2.0 and set InheritDocTrimLevel=private (#17429)

kasobol-msft committed:
add retry when booting azurite. (#17430)

kinelski committed:
[MetricsAdvisor] Implemented tests for the Feedback API (#17396)

mikeharder committed:
[Perf, Stress] Downgrade Azure.Core dependency (#17437)

- Frameworks should depend on lowest version of Azure.Core for maximum compat

azure-sdk committed:
Clean up modules direcotry in eng/common since they are no longer being used. (#17301)

Co-authored-by: Chidozie Ononiwu <chononiw@microsoft.com>

azabbasi committed:
Wait 30 seconds after deployment is done. (#17431)

Commit:
Azure Monitor Exporter: new test to verify correlation ids using Activity + iLogger (#17189)

* new test to verify correlation ids using Activity + iLogger

* dispose tracerProvider

* de-dupe exporter options

* cleanup

* dedupe activitysourcename

* additional activity test

jsquire committed:
[Event Hubs Client] Remove Code Analysis Supressions (#17428)

The focus of these changes is to remove the code analysis rule suppressions
added as a temporary measure during the introduction of additional rules.

christothes committed:
Update changelog for release (#17450)

chamons committed:
[Synapse] Add create/execute pipeline sample (#17282)

azure-sdk committed:
Increment version for tables releases (#17452)

Increment package version after release of Azure.Data.Tables

seanmcc-msft committed:
Fixed Storage Open Read Seek bug. (#17443)

maririos committed:
keyphrases tests (#17441)

LarryOsterman committed:
Fixed moved file (#17462)

azure-sdk committed:
Create PR to target against master (#17456)

Co-authored-by: Sima Zhu <sizhu@microsoft.com>

pakrym committed:
Re-enable ApplicationInsights tests (#17463)

Remove the distributes testing code that is not required anymore since Azure.Messaging.EventHubs handles more of the.

Bring back the ApplicationInsights tests.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17205

kasobol-msft committed:
[Storage] Parallelize live tests. (#17454)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/6905

In this PR:
- introduce LiveParallizable attribute.
- changed scope of DiagosticScopeInterceptor as it's not thread safe, see:
    - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=649697&view=results
    - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=649767&view=results

pratimaupadhyay02 committed:
SDK upgrade for package 2020-10-01 (#17304)

* SDK update for package 2020-10-01

* Adding new files

* Update

* Update session records

* Dummy check-in

* Updating version and release notes

Commit:
[Identity] Fix issue of discarding content headers (#17467)

Fixes #17448

jorgecotillo committed:
[Microsoft.Azure.Management.ResourceManager] [DeploymentScripts] New Api-version (#17170)

* bump resources api version

* added new models

* updated recordings

* Metadata file

* Update AzSdk.RP.props

Co-authored-by: Gokul Premraj <gopremra@microsoft.com>

suhas92 committed:
[Text Analytics] Add samples for manual polling  (#17444)

This PR adds 2 new samples for manual polling for AnalyzeOperation and HealthcareOperation.

NOTE: User can use `UpdateStatusAsyc()` method with a `pollingInterval` passed as an argument. If the user wants to have an automatic polling by default, `WaitForCompletionAsync()` method should be used.

maririos committed:
[TA] Clean up samples (#17426)

* clean up samples

* English

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1270 (#17479)

* resolve git longpath error when applying documentation updates

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>

maririos committed:
[TA] Remove old snippet reference (#17478)

`TextAnalyticsSample4CreateClient` was removed and now TA samples is using `CreateTextAnalyticsClient`.
This wasn't fixed in the latest sample addition, so fixing it.

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1274 (#17477)

* Added a step of skipping package json update for spring boot packages

* Added skip package json config

* Update the indentation

Co-authored-by: Sima Zhu <sizhu@microsoft.com>

praveenkuttappan committed:
Test Auto API review change (#17418)

*Auto create API review

azabbasi committed:
Fix(pipeline) Wait 1 minute for permissions to propagate (#17490)

* Wait 5 minutes for permissions to propagate

* Update test-resources-post.ps1

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1273 (#17487)

* Support alpha and azure pipeline build version formats for SemVer parsing and sorting

* Minor build number fixes

* Support zero-padding of build versions more generically in SemVer script

* Fix pre-release label conditional for HasValidPrereleaseLabel in semver script

* Set default convention for build number separator in semver script

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

LarryOsterman committed:
Updated version number (#17464)

* Updated version number

* Try #2 on beta.1

* Fixed typo in version number

chamons committed:
[Synapse] Fix/Disable Tests so they consistently pass (#17460)

- Disable a number of tests with https://github.com/Azure/azure-sdk-for-net/issues/17455 as a tracking issues 
- Update test-resources.json so you can run tests with preconfigured resources
- "Smash" some test sets into single test so snippets run in correct consistent order
- Update trigger tests to use derived trigger type (so the name comes back correctly). 
   - Digging into why this is necessary is tracked in https://github.com/Azure/azure-sdk-for-net/issues/17459
- I also removed some unnecessary/unused snippet regions

jsquire committed:
[Event Hubs Client] Test Prefetch Adjustments (#17433)

The focus of these changes is to tweak configuration of the Event Hubs
clients in scenarios where the behavior of multiple readers acting on the
same partition are being validated.  These are susceptible to a race condition
where the prefetch queue size was large enough that all expected events
were held and the expected service calls did not take place, causing
intermittent test failures.

pakrym committed:
Remove the need to specify service directory for the test environment (#17476)

Now that we have the `SourcePath` attribute on all test assemblies the `TestEnvironment` usage can be simplified.

maririos committed:
[FR] Don't use inheritdoc for now (#17506)

* don't use inheritdoc

* remove unnecessary inheritdoc

pakrym committed:
Prepare for Azure.Core 1.7.0 release (#17503)

kasobol-msft committed:
Improve docs around encoding and previous version behavior. (#17436)

mitchdenny committed:
Switch to azsdk-pool-mms-win-2019-general pool. (#17515)

kasobol-msft committed:
sometimes cancellation doesnt fire immediatelly. (#17512)

maririos committed:
[TA] cleanup on detect language tests (#17466)

* detect language

* PR feedback

* add missing validation for warnings

hivyas committed:
[Media] LVA SDK addition (#17289)

Co-authored-by: giakas <giakas@microsoft.com>

eliagrady committed:
add Source vs. source discriminator customizations (#17288)

add Assessments and SubAssessments tests + sessionRecords

add Assessments CreateOrUpdate test + sessionRecords

changing custom class PolymorphicJsonCustomConverter to 'internal'

changing non-overriding methods in PolymorphicJsonCustomConverter to 'internal'

adding newline at <EOF>

Co-authored-by: full name <elgrady@microsoft.com>

giakas committed:
[LVA]Adding the missing project in ci.yml (#17523)

seanmcc-msft committed:
Add connection string constructors to Data Lake (#17511)

Mohit-Chakraborty committed:
Add perf tests for uploading and downloading files for Azure Storage Files Shares (#17453)

* Add tests for uploading and downloading files

kinelski committed:
[MetricsAdvisor] Add tests for the Hook CRUD operations (#17489)

maririos committed:
Add Suhas to TA codeowners (#17518)

pakrym committed:
Update Azure.Core package version (#17519)

pakrym committed:
Prepare to produce a nightly build of EventHubs extension (#17520)

We need a package ref to Microsoft.Extensions.Azure

azabbasi committed:
ADT - livetest - Wait 5 minutes after role assignment and before the test run to avoid 403s. (#17539)

hivyas committed:
[Media] Removing periods from safename (#17537)

kasobol-msft committed:
[Storage] Bump threadpool in tests. (#17547)

dochung4 committed:
Make DedicatedHostGroup.properties.supportAutomaticPlacement default to false (#17224)

pakrym committed:
Fix Microsoft.Azure.WebJobs.Extensions.EventHubs inheritdoc failures (#17551)

Mohit-Chakraborty committed:
Generate 'SourcePath' attribute in Perf test assemblies (#17545)

* Generate 'SourcePath' attribute in Perf test assemblies

* Include stress projects in the condition to add the attribute

JoshLove-msft committed:
Include Lease Info in logging (#17509)

Fixes https://github.com/Azure/azure-sdk-for-net/issues/17032

JoshLove-msft committed:
Ensure that type is populated when deserializing into a CloudEvent (#17468)

During UX studies, users were easily confused if they attempted to deserialize an event grid event into a CloudEvent as no exception was thrown. We intentionally removed the validation as part of addressing https://github.com/Azure/azure-sdk-for-net/issues/16012, but this may have gone too far in the other direction. A good compromise seems to leave in the validation that the service uses which is to ensure that Type is populated for CloudEvents, while allowing Source to be null.

pakrym committed:
Remove explicit package references on net5.0 (#17544)

crtreasu-msft committed:
Added pipelines for Mixed Reality SDKs (#17554)

* This change adds pipelines for Mixed Reality services

* Fixed PR path

* Removed TestPipeline: true

Co-authored-by: Craig Treasure <crtreasu@microsoft.com>

pakrym committed:
Fix the flakiness in secrets extension test (#17559)

aim-for-better committed:
[HDInsight] Support extending list hosts api and computeisolation properties. (#17532)

* Support extending list hosts api and computeisolation properties.

* Fix test errors

Co-authored-by: Zhenyu Zhou <zhezhou@microsoft.com>

pavelprystinka committed:
[Communication] - PhoneNumberAdministration - add tests (#17492)

* Add PhoneNumberAdministration tests

pakrym committed:
Prepare Azure.Extensions.AspNetCore.DataProtection.Blobs for release (#17570)

ramya-rao-a committed:
[Service Bus] Use docs.microsoft.com link for API references (#17577)

azure-sdk committed:
Fixed the bug of replacing img src with href text (#17586)

Co-authored-by: Sima Zhu <sizhu@microsoft.com>

minnieliu committed:
[Communication] - All - Rename CommunicationUserCredential to CommunicationTokenCredential (#17553)

* Rename CommunicationUserCredential to CommunicationTokenCredential

* Update changelog

* Update readme

* Update readme

* Renaming bearer test file

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

chamons committed:
[synapse] Add two new champion scenario tests, refactor one, and disable a few (#17546)

- Add beginnings of two new champion scenario tests (Upload artifact notebook and monitoring pipelines)
  - Once all of the tests are in they will be turned into snippets and made markdown
  - Significant amount of polling code in Sample2_ExecuteSparkStatement.cs is from service team recommendation right now
      - Moving it into the binding as a LRO is a future research topic
      - Tracked in https://github.com/Azure/azure-sdk-for-net/issues/17587 to track the copied code. 
- Refactor CreateAndRunPipeline test which had two tests, which were order dependent :facepalm: 
- Disable a few more tests that now fail in LIVE (will be reenabled when the [swagger PR](https://github.com/Azure/azure-rest-api-specs/pull/12093) lands and is bumped to)

azure-sdk committed:
Increment version for extensions releases (#17592)

Increment package version after release of Azure.Extensions.AspNetCore.DataProtection.Blobs

seanmcc-msft committed:
Feature/storage/data lake set permission bug (#17591)

jaschrep-msft committed:
Fixed broken docs link, (#17593)

Points developers to the overload we want them to use, rather than the
hidden one we "retired."
Likely solves an MS Docs rendering bug, where it can't link to the
method properly because methods we hide from intellisense also aren't
rendered to MS Docs at all, so there's nothing to link to.

Co-authored-by: jschrepp-MSFT <41338290+jschrepp-MSFT@users.noreply.github.com>

dhawal777 committed:
Updating release notes. (#17525)

* Updating release notes to highlight additional changes that came as part of last update

* Spell fix

seanmcc-msft committed:
Fixed bug where BlobBaseClient.Exists() did not function correctly for blob encrypted with Customer Provided Key or Encryption Scope. (#17599)

jaschrep-msft committed:
Misc blobs migration guide fixes (#17601)

* Misc blobs migration guide fixes

Fixed some reported ambiguities and rendering bugs.

* Fixed variable name in sample project

* build error

Co-authored-by: jschrepp-MSFT <41338290+jschrepp-MSFT@users.noreply.github.com>

JoshuaLai committed:
[Communication] - Identity - Auth with Token Credential (#17438)

* [CommunicationIdentityClient] adding in new method to create beaer token. Fixing spelling mistake.

* [CommunicationIdentityClient] adding new init passing in token credential

* [TokenCredential] finalize the parameters to take in a token credential and an endpoint string

* [ClientOptionsExtensions] removing the scope from the BearerTokenAuth policy

* [CommunicationIdentityClientLiveTests] Adding new live test init with token credential

* [PhoneNumberAdministrationClient] revert the init with token

* [CommunicationIdentityClient] passing in endpoint as an uri

* [CommunicationIdentityClient] assert on the endpoint

* [CommunicationIdentityClient] reorder of the parameters

* Adding working live tests

* [Scripts] New generated code

* [CommunicationIdentityClient] update the comment for tokenCredential to be accurate

* [CommunicationIdentityClient] update the comment for endpoont to be more accurate

* [CommunicationIdentityClientLiveTests] Nit on the the test name

* Update CHANGELOG.md

* [IdentityClientFromTokenWithMultipleScopes] rename the generated files

* [CommunicationRecordedTestSanitizer] adding sanitizer to COMMUNICATION_ENDPOINT_STRING

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

pakrym committed:
Auto create storage container (#17607)

pakrym committed:
Organize Packages.Data.props (#17603)

This PR splits `Packages.Data.props` into strict groups:

1. Track 1 dependencies
2. Track 2 dependencies
3. Track 2 extended dependencies (Microsoft.Azure.* T2 libraries)
4. Build time dependencies
5. Test dependencies

Some versions are duplicated across groups, this is intentional so someone trying to update test package version doesn't change the shipping package dependency.

pakrym committed:
Prepare Azure.Extensions.AspNetCore.DataProtection.Blobs for release again (#17610)

pakrym committed:
Add EventHub extension README.md (#17514)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/16646

AlexGhiondea committed:
Update CODEOWNERS (#17605)

The TimeSeriesInsights label was incorrect. Fixing it.

Also ensure that we label PRs in the Synapse folder as well.

azure-sdk committed:
Increment version for extensions releases (#17627)

Increment package version after release of Azure.Extensions.AspNetCore.DataProtection.Blobs

JoshLove-msft committed:
Authorization refresh fix (#17609)

* save

* Authorization refresh fix

* Undo inadvertent changes

minnieliu committed:
[Communication] - Common - Adding Identifier Suffix to all Identifier types (#17608)

* Adding Identifier Suffix to all Identifier types

* Update generated files

* Fix minor issues

* Update changelog

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

JoshuaLai committed:
[COMMUNICATION] - [LIVE TESTS] - adding missing env var (#17631)

* [test-resource] adding in new environment variable that was missed for CI live test

* [resource] encoded space

Mohit-Chakraborty committed:
Log exceptions from the failed step in perf test environment (#17629)

OlegSternbergIBM committed:
Audit devops sdk (#17529)

* generated sdk + tests

* Update version

* Stage test passes. Both files need to be adjusted for Prod, once API will be exposed there.

* consiume new DevOps swagger

* validated DevOps in Prod

* fix version

* update recorded test file

* updated version to 1.48

Co-authored-by: Oleg Sternberg <olsternb@microsoft.com>

crtreasu-msft committed:
Added the Azure.MixedReality.Authentication library (#17571)

This change adds an authentication library for Mixed Reality services, which all currently utilize a custom STS for authentication. Upcoming Mixed Reality client libraries will depend on this library to perform authentication using identities from Azure.Identity. See the added `README.md` for more information about the library.

[ApiView link](https://apiview.dev/Assemblies/Review/35afdb6afa874f6ab4caa1cfaf7a15cd)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1287 (#17630)

* Move common code to create API review into eng common in tools

azure-sdk committed:
Update AutoRest C# version (#17516)

Update AutoRest C# version to 3.0.0-beta.20201217.1

pakrym committed:
Add System.IdentityModel.Tokens.Jwt to versions (#17637)

chjaress committed:
Add tests for enableAutomaticUpgrade property for VM Extensions (#17604)

* Adding VM Extension test and session records for tests

* Adding test for VMScaleSetVMExtensions

* Reverting some generated files

ShivangiReja committed:
Update "AutoRest.CSharp.V3" package name to "AutoRest.CSharp" (#17638)

Update "AutoRest.CSharp.V3" package name to "AutoRest.CSharp"

pakrym committed:
Add AsyncPageable scope validation (#17635)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17633

minnieliu committed:
[Communication] - Common -  Adding TokenRefreshOptions to CommunicationTokenCredential ctor (#17597)

* Rename CommunicationUserCredential to CommunicationTokenCredential

* Update changelog

* Update readme

* Update readme

* Encapsulate CommunicationUserCredential params into TokenRefreshOptions

* Renaming bearer test file

* Address previous PR's comments

* Update generated files

* Removing Identity namespace

* Fix format

* Update changelog

* Update Merge conflict

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

Mohit-Chakraborty committed:
Add perf tests for MS Azure File Storage (T1) API (#17555)

* Add perf tests for MS Azure File Storage (T1) API

chamons committed:
[synapse] Re-enable many disabled tests by bumping to swagger with nullability fix (#17645)

- This fixes almost all of https://github.com/Azure/azure-sdk-for-net/issues/17455
- Two tests remain: A managed private endpoint issue that's been raised w\ the service team and a pipeline test that I'm rewriting in another PR

JoshLove-msft committed:
Rename VolatileQueryParameters to IgnoredQueryParameters and update behavior of ExcludeHeaders (#13403)

azure-sdk committed:
Update AutoRest C# version (#17643)

* Update AutoRest C# version to 3.0.0-beta.20201217.3

kinelski committed:
[MetricsAdvisor] Added tests for the AnomalyDetectionConfiguration CRUD operations (#17590)

chamons committed:
[synapse] Convert samples to markdown format (#17657)

- Converted all synapse samples (both per-existing and those that I wrote) to markdown + cs snippet format
- First pass at documenting the 4-5 that are champion scenarios
- Found and deleting some duplicate snippet + scenarios I found along the way.

chamons committed:
[TestFramework] Add README note about sharing fields and tests (#17663)

jsquire committed:
[Docs] Add PowerShell to Contributing Guide (#17689)

The focus of these changes is to highlight the PowerShell 6/7 dependency
when working with the engineering scripts, such as generating the API listing
or updating code snippets.

chamons committed:
[synapse] Fix many README links (#17682)

- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17666
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17667
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17668
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17669
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17670

DominikMe committed:
Add communication identifier for Microsoft Teams user (#17675)

nokternol committed:
#17662 [bug] add setters to LengthTokenilter (#17694)

* #17662 [bug] add setters to LengthTokenilter

* #17662 Run Export-API.ps1

Co-authored-by: mark butterworth <mark.butterworth@carestream.com>

Luyunmt committed:
resolve search problem

Luyunmt committed:
resolve search problem

Luyunmt committed:
change formrecognizer test yml

Luyunmt committed:
change formrecognizer test yml

Luyunmt committed:
change text analyze resourcegroup

Luyunmt committed:
change text analyze resourcegroup

Luyunmt committed:
change subscriptionconfigureation

Luyunmt committed:
change subscriptionconfigureation

Luyunmt committed:
change subscriptionconfigureation

Luyunmt committed:
change subscriptionconfigureation

Luyunmt committed:
change identity test yml

Luyunmt committed:
change identity test yml

Luyunmt committed:
change identity code

Luyunmt committed:
change identity code

maririos committed:
[TA] Assigning canary to centraluseuap (#17517)

* assigning canary to centraluseuap

* override the whole thing

maririos committed:
Increase TimeoutInMinutes for FR (#17709)

kinelski committed:
[MetricsAdvisor] Added tests for the AnomalyAlertConfiguration CRUD operations (#17671)

maririos committed:
enable receipts (#17722)

Luyunmt committed:
rebase master

Luyunmt committed:
resolve conflicts

Luyunmt committed:
resolve conflicts

Luyunmt committed:
change formrecognizer and suspcritionconfiguration

Luyunmt committed:
change storage

Luyunmt committed:
change storage and textanalytics

Luyunmt committed:
removing some unnecessary change

jsquire commented:
Hi @Luyunmt.  I believe this has been obviated with Ben's recent PR.   Since there hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream by removing the `no-recent-activity` label?  Otherwise, we'll close this out in 7 days.

jsquire commented:
Since there hasn't been recent engagement, this is being closed out.  Please feel free to reopen if you'd like to continue working on these changes.
